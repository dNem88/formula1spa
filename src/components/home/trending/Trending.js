import React  from 'react';
import styles from './trending.module.css'
import {useNavigate} from 'react-router-dom'

function Trending(props) {
    console.log(props.article)
    let navigate = useNavigate();
    function clickHandler(e) {
        navigate(`/news/${props.article._id}`)
    }
    return (
        <div id={props.article._id} onClick={clickHandler} className={styles['wrapper']}>
            <p className={styles['trending-p']}>TRENDING</p>
            <h1 className={styles['trending-heading']}>{props.article.title}</h1>
            <div className={styles['image-container']}>
                <img className={styles['trending-image']} src={props.article.urlToImage} alt='someAlt'/>
            </div>
        </div>
    )
}
export default Trending