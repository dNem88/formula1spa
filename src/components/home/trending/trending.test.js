import React from 'react'
import renderer from 'react-test-renderer'
import Trending from './Trending'
import {MemoryRouter} from 'react-router-dom'

describe('Trending component', () => {
    
    it('Testing trendin', () => {
        let trending = renderer.create(
        <MemoryRouter>
            <Trending article={{title: 'test-title', _id: 'test-id', urlToImage: 'test-url'}}/>
        </MemoryRouter>
    ).toJSON() 
        expect(trending).toMatchSnapshot()
        expect(trending.props.id).toBe('test-id')
        expect(trending.props.className).toBe('wrapper')
        expect(trending.type).toBe('div')
        expect(trending.children[0].props.className).toBe('trending-p')
        expect(trending.children[0].type).toBe('p')
        expect(trending.children[0].children[0]).toBe('TRENDING')
        expect(trending.children[1].props.className).toBe('trending-heading')
        expect(trending.children[1].type).toBe('h1')
        expect(trending.children[1].children[0]).toBe('test-title')
        expect(trending.children[2].props.className).toBe('image-container')
        expect(trending.children[2].type).toBe('div')
        expect(trending.children[2].children[0].props.className).toBe('trending-image')
        expect(trending.children[2].children[0].type).toBe('img')
        expect(trending.children[2].children[0].props.src).toBe('test-url')
        
    })
})