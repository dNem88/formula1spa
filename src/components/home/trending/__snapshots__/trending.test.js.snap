// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending component Testing trendin 1`] = `
<div
  className="wrapper"
  id="test-id"
  onClick={[Function]}
>
  <p
    className="trending-p"
  >
    TRENDING
  </p>
  <h1
    className="trending-heading"
  >
    test-title
  </h1>
  <div
    className="image-container"
  >
    <img
      alt="someAlt"
      className="trending-image"
      src="test-url"
    />
  </div>
</div>
`;
